libs/api/ebpf_api.cpp

```diff
@@ -2280,20 +2280,18 @@ ebpf_program_load_bytes(
     load_info.program_type = *program_type;
     load_info.program_handle = reinterpret_cast<file_handle_t>(program_handle);
     load_info.execution_type = execution_type;
-    load_info.byte_code = const_cast<uint8_t*>(byte_code);
-    load_info.byte_code_size = byte_code_size;
+    load_info.instructions = (ebpf_instruction_t*)instructions;
+    load_info.instruction_count = instruction_count;
     load_info.execution_context = execution_context_kernel_mode;
 
     // Resolve map handles in byte code.
     std::vector<original_fd_handle_map_t> handle_map;
-    const ebpf_inst* instructions = reinterpret_cast<const ebpf_inst*>(byte_code);
-    const ebpf_inst* instruction_end = reinterpret_cast<const ebpf_inst*>(byte_code + byte_code_size);
-    for (size_t index = 0; index < byte_code_size / sizeof(ebpf_inst); index++) {
+    for (size_t index = 0; index < instruction_count; index++) {
         const ebpf_inst& first_instruction = instructions[index];
         if (first_instruction.opcode != INST_OP_LDDW_IMM) {
             continue;
         }
-        if (&instructions[index + 1] >= instruction_end) {
+        if (index + 1 >= instruction_count) {
```

Comment:
since this is in a for loop and I dont see any other place where `index` is incremented, is this check needed any longer?

Reply:
Not sure offhand, but it's not a change introduced in this PR.